docs/coding-guidelines/vectorization-guidelines.md

```diff
@@ -0,0 +1,1091 @@
+
+    uint mostSignificantBits = equals.ExtractMostSignificantBits();
+    int index = BitOperations.TrailingZeroCount(mostSignificantBits);
+    
+    return elementOffset + index;
+}
+```
+
+### Comparison
+
+Beside equality checks, vector APIs allow for comparison. The `bool`-returning overloads return `true` when the given condition is true:
+
+```cs
+public static bool GreaterThanAll<T>(Vector128<T> left, Vector128<T> right) where T : struct
+public static bool GreaterThanAny<T>(Vector128<T> left, Vector128<T> right) where T : struct
+public static bool GreaterThanOrEqualAll<T>(Vector128<T> left, Vector128<T> right) where T : struct
+public static bool GreaterThanOrEqualAny<T>(Vector128<T> left, Vector128<T> right) where T : struct
+public static bool LessThanAll<T>(Vector128<T> left, Vector128<T> right) where T : struct
+public static bool LessThanAny<T>(Vector128<T> left, Vector128<T> right) where T : struct
+public static bool LessThanOrEqualAll<T>(Vector128<T> left, Vector128<T> right) where T : struct
+public static bool LessThanOrEqualAny<T>(Vector128<T> left, Vector128<T> right) where T : struct
+```
+
+Similarly to `Equals`, vector-returning overloads return a vector whose elements are all-bits-set or zero, depending on whether the corresponding elements in `left` and `right` meet the given condition.
+
+```cs
+public static Vector128<T> GreaterThan<T>(Vector128<T> left, Vector128<T> right) where T : struct
+public static Vector128<T> GreaterThanOrEqual<T>(Vector128<T> left, Vector128<T> right) where T : struct
+public static Vector128<T> LessThan<T>(Vector128<T> left, Vector128<T> right) where T : struct
+public static Vector128<T> LessThanOrEqual<T>(Vector128<T> left, Vector128<T> right) where T : struct
+```
+
+`ConditionalSelect` Conditionally selects a value from two vectors on a bitwise basis.
+
+```cs
+public static Vector128<T> ConditionalSelect<T>(Vector128<T> condition, Vector128<T> left, Vector128<T> right)
+```
+
+This method deserves a self-describing example:
+
+```cs
+Vector128<float> left = Vector128.Create(1.0f, 2, 3, 4);
+Vector128<float> right = Vector128.Create(4.0f, 3, 2, 1);
+
+Vector128<float> result = Vector128.ConditionalSelect(Vector128.GreaterThan(left, right), left, right);
+
+Assert.Equal(Vector128.Create(4.0f, 3, 3, 4), result);
+```
+
+### Math
+
+Very simple math operations can be also expressed by using the operators:
+
+```cs
+public static Vector128<T> Add<T>(Vector128<T> left, Vector128<T> right) where T : struct => left + right;
+public static Vector128<T> Divide<T>(Vector128<T> left, Vector128<T> right) => left / right;
+public static Vector128<T> Divide<T>(Vector128<T> left, T right) => left / right;
+public static Vector128<T> Multiply<T>(Vector128<T> left, Vector128<T> right) => left * right;
+public static Vector128<T> Multiply<T>(Vector128<T> left, T right) => left * right;
+public static Vector128<T> Subtract<T>(Vector128<T> left, Vector128<T> right) => left - right;
```

Comment:
Same as above, the operators should be preferred where possible, as it helps avoid bugs around operator precedence and can improve readability.